Sources/AardvarkReveal/ApplicationState.swift

```diff
@@ -0,0 +1,183 @@
+//
+//  Copyright 2020 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+enum Class: Equatable {
+
+    case baseClass(name: String)
+
+    indirect case subclass(name: String, superclass: Class)
+
+    // MARK: - Internal Methods
+
+    func isTypeOf(_ className: String) -> Bool {
+        switch self {
+        case let .baseClass(name: name):
+            return name == className
+
+        case let .subclass(name: name, superclass: superclass):
+            return name == className || superclass.isTypeOf(className)
+        }
+    }
+
+}
+
+extension Class: Codable {
+
+    init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        let name = try values.decode(String.self, forKey: .name)
+        if let superclass = try? values.decode(Class.self, forKey: .superclass) {
+            self = .subclass(name: name, superclass: superclass)
+        } else {
+            self = .baseClass(name: name)
+        }
+    }
+
+    func encode(to encoder: Encoder) throws {
+        var values = encoder.container(keyedBy: CodingKeys.self)
+        switch self {
+        case let .baseClass(name: name):
+            try values.encode(name, forKey: .name)
+        case let .subclass(name: name, superclass: superclass):
+            try values.encode(name, forKey: .name)
+            try values.encode(superclass, forKey: .superclass)
+        }
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case name = "n"
```

Comment:
If this case name is required due to compatibility with another library or model, would be good to state as much in a comment

Reply:
Added a comment at the top of the file. This whole file mirrors the format used in the JSON representation of the application state included in a `.reveal` file.